hack/tests/Makefile

```diff
@@ -0,0 +1,202 @@
+DEV_REGISTRY ?= localhost:5000
+
+.start-kind-cluster:
+	kind delete cluster; sleep 20; docker network create kind; cd $(TEST_DIR)/utils; ./kind-with-registry.sh
+
+## --------------------------------------
+## Build CAPI
+## --------------------------------------
+
+# Ask the user for the CAPI path, if not provided
+.ensure-capi-path:
+ifndef CAPI_PATH
+	$(eval export CAPI_PATH = $(shell read -p $$'🔖 \e[38;5;208mPlease specify the CAPI_PATH:\e[0m ' capi_path; echo $$capi_path))
+	@printf "\033[2K\r"
+endif
+
+## --------------------------------------
+## Configure clusterctl
+## --------------------------------------
+
+define CLUSTERCTL_SETTINGS_PROVIDERS
+{
+    "providers": ["cluster-api", "bootstrap-kubeadm", "control-plane-kubeadm", "infrastructure-kubemark", "infrastructure-docker"],
+    "provider_repos": ["$(ROOT_DIR)"]
+}
+endef
+
+## --------------------------------------
+## Build clusterctl
+## --------------------------------------
+
+.capi-build-clusterctl: .ensure-capi-path
+	@cd $(CAPI_PATH) && make clusterctl
+
+.generate-clusterctl-config: export CLUSTERCTL_SETTINGS:=$(CLUSTERCTL_SETTINGS_PROVIDERS)
+.generate-clusterctl-config: .ensure-capi-path
+	@rm -f $(CAPI_PATH)/clusterctl-settings.json
+	@echo "$${CLUSTERCTL_SETTINGS}" >> $(abspath ${CAPI_PATH})/clusterctl-settings.json
+
+
+## --------------------------------------
+## Configure kubemark clusterctl
+## --------------------------------------
+
+define CLUSTERCTL_SETTINGS_CAPK
+{
+    "name": "infrastructure-kubemark",
+    "config": {
+        "componentsFile": "infrastructure-components.yaml",
+        "nextVersion": "v0.2.0"
+    }
+}
+endef	
+
+.generate-kubemark-config: export CLUSTERCTL_SETTINGS:=$(CLUSTERCTL_SETTINGS_CAPK)
+.generate-kubemark-config:
+	@rm -f $(ROOT_DIR)/clusterctl-settings.json
+	@echo "$${CLUSTERCTL_SETTINGS}" >> $(abspath ${ROOT_DIR})/clusterctl-settings.json
+
+
+.docker-build-capi: .ensure-capi-path
+	@make -C $(CAPI_PATH) docker-build REGISTRY=$(DEV_REGISTRY) && \
+		make -C $(CAPI_PATH) docker-push REGISTRY=$(DEV_REGISTRY)
+	@make -C ${CAPI_PATH}/test/infrastructure/docker docker-build REGISTRY=$(DEV_REGISTRY) && \
+  	make -C ${CAPI_PATH}/test/infrastructure/docker docker-push REGISTRY=$(DEV_REGISTRY)
+
+.docker-build-capk:
+	@cd $(ROOT_DIR) && make release-manifests
+	@make -C $(ROOT_DIR) docker-build REGISTRY=$(DEV_REGISTRY)
+	@cp $(ROOT_DIR)/out/infrastructure-components.yaml ~/.cluster-api/dev-repository/infrastructure-kubemark/v0.2.0 
+	@cp $(ROOT_DIR)/templates/cluster-template-capd.yaml ~/.cluster-api/dev-repository/infrastructure-kubemark/v0.2.0 
+	@cp $(ROOT_DIR)/templates/cluster-template.yaml ~/.cluster-api/dev-repository/infrastructure-kubemark/v0.2.0  
+	@cd $(HOME)/.cluster-api/dev-repository/infrastructure-kubemark/v0.2.0/
+	@perl -i -l -p -e 'print "        - --kubemark-image=quay.io/elmiko/kubemark" if $$. == 428' $(HOME)/.cluster-api/dev-repository/infrastructure-kubemark/v0.2.0/infrastructure-components.yaml
```

Comment:
this replacement line is not working for me, it produces an invalid output in the yaml file, i see this:

```
- apiGroups:
  - infrastructure.cluster.x-k8s.io
  resources:
        - --kubemark-image=quay.io/elmiko/kubemark
  - kubemarkmachines/status
  verbs:
  - get
  - patch
  - update
```